### PR TITLE
Proposal encoding documentation, split encoding into multiple files

### PR DESCRIPTION
**Pull request by Siegrift**
Fixes https://github.com/api3dao/api3-dao-dashboard/issues/221
Fixes https://github.com/api3dao/api3-dao-dashboard/issues/220
Fixes https://github.com/api3dao/api3-dao-dashboard/issues/219

### CODE REVIEW

**Comment by bbenligiray** [comment]
Nice. I was thinking of pure documentation but this is probably more useful.

File: src/logic/proposals/encoding/encoding.ts [hunk 1]
```diff
@@ -1,42 +1,46 @@
+/**
+ * When a proposal is passed, it can be used to execute the EVM script that was specified when the proposal was created.
+ * The script can be a function call or multiple function calls. Proposal needs to be executed by separate transaction
+ * and the executor pays gas also for the EVM script that is run.
```
Comment (andreogle) [inline src/logic/proposals/encoding/encoding.ts:4]:
```suggestion
 * and the executor will pay the gas cost for the EVM script that is run.
```

File: src/logic/proposals/encoding/encoding.ts [hunk 2]
```diff
@@ -1,42 +1,46 @@
...
+ *
+ * For now, we only support a single function call in the EVM script. The reasons are that it's enough for the most
+ * cases, it's easier to encode it and we can create simpler UI for the users.
```
Comment (andreogle) [inline src/logic/proposals/encoding/encoding.ts:7]:
```suggestion
 * cases, it's easier to encode and we can create a simpler UI for end users.
```

File: src/logic/proposals/encoding/encoding.ts [hunk 3]
```diff
@@ -46,11 +50,21 @@ type EncodedEvmScriptError = {
...
+ * @param provider a provider which is able to resolve ENS names
+ * @param formData the proposal data to be encoded
+ * @param api3Agent the addresses of the voting app agents
+ */
 export const encodeEvmScript = async (
   provider: providers.Provider,
   formData: NewProposalFormData,
   api3Agent: Api3Agent
 ): Promise<EncodedEvmScript> => {
+  // Ensure that the form parameters form a valid JSON
```
Comment (andreogle) [inline src/logic/proposals/encoding/encoding.ts:67]:
```suggestion
  // Ensure that the form parameters form a valid JSON array
```

**Review by andreogle — APPROVED** [review]
Thanks for this! Just some minor nit comments about the way some comments are worded.

**Pull request merged**